test/test_c10d.py

```diff
@@ -2831,6 +2848,83 @@ def _test_broadcast_coalesced(self, process_group, device):
 
         self.assertEqual(tensors, target)
 
+    def _run_all_reduce(self, pg):
+        pg.allreduce(torch.rand(10).cuda(self.rank))
+
+    @requires_nccl()
+    @skip_if_not_multigpu
+    def test_nccl_errors_nonblocking(self):
+        store = c10d.FileStore(self.file.name, self.world_size)
+        process_group = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
+        process_group.allreduce(torch.rand(10).cuda(self.rank))
+        if self.rank == 0:
+            # Should run fine without raising any errors.
```

Comment:
Let's be more explicit in the comment. IIUC, shall we add something like: This `allreduce` does not block Python thread as `allreduce` enqueues the cuda operation, and then `wait` only blocks the current cuda stream.